inc/pkthlp.h

```diff
@@ -308,6 +309,125 @@ PktBuildTcpFrame(
     return TRUE;
 }
 
+//
+// Build a QUIC frame from the given parameters and paylaod.
+// The frame is meant to be wraped in a TCP or UDP frames,
+// as the payload in `PktBuildTcpFrame` or `PktBuildUdpFrame`.
+//
+inline
+_Success_(return != FALSE)
+BOOLEAN
+PktBuildQuicLongHeader(
+    _Out_writes_bytes_(*BufferSize) VOID *Buffer,
+    _Inout_ UINT32 *BufferSize,
+    _In_ UINT8 TypeAndSpecificBits,
+    _In_reads_bytes_(DestConnIdLength) CONST UCHAR *DestConnId,
+    _In_ UINT8 DestConnIdLength,
+    _In_reads_bytes_(SrcConnIdLength) CONST UCHAR *SrcConnId,
+    _In_ UINT8 SrcConnIdLength,
+    _In_opt_ CONST UCHAR *Payload,
+    _In_ UINT16 PayloadLength
+    )
+{
+    UINT8* Cursor = (UINT8*)Buffer;
+    CONST UINT32 Version = 1;
+    CONST UINT32 TotalLength = sizeof(UINT8) + sizeof(Version) + sizeof(DestConnIdLength) +
+        DestConnIdLength + sizeof(SrcConnIdLength) + SrcConnIdLength + PayloadLength;
+
+    if (*BufferSize < TotalLength) {
+        return FALSE;
+    }
+
+    //
+    // Set the Header Form bit to 1 for a long header, and Fixed bit to 1
+    //
+    CONST UINT8 Prelude = 0b1100'0000 | (0b0011'1111 & TypeAndSpecificBits);
```

Comment:
Is this `0b` and `'` syntax backwards compatible with old C compilers? I think this is a newer feature that, while nice, could cause compat problems if reused by other compilers.

Reply:
Hum, looking around, binary literals are a largely supported GNU extension that made its way in C23.
And the ' is C23.
That's more recent than I thought, it's been there for 10+ years in C++! I'll move to hexadecimal.

Reply:
Yeah, C gets all the features backported after 10 years of testing in C++ ;)